spring-kafka/src/test/java/org/springframework/kafka/core/KafkaTemplateTests.java

```diff
@@ -265,6 +267,29 @@ public boolean isInterestedInSuccess() {
 	}
 
 	@Test
+	public void withCallback() throws Exception {
+		Map<String, Object> senderProps = KafkaTestUtils.producerProps(embeddedKafka);
+		DefaultKafkaProducerFactory<Integer, String> pf = new DefaultKafkaProducerFactory<>(senderProps);
+		KafkaTemplate<Integer, String> template = new KafkaTemplate<>(pf);
+		template.setDefaultTopic(INT_KEY_TOPIC);
+		final CountDownLatch latch = new CountDownLatch(1);
+		template.send(new ProducerRecord<Integer, String>(INT_KEY_TOPIC, "foo"), new Callback() {
+
+			@Override
+			public void onCompletion(RecordMetadata metadata, Exception exception) {
+				latch.countDown();
+			}
+
+		});
+		template.flush();
+		assertThat(latch.await(10, TimeUnit.SECONDS)).isTrue();
+
+		//Drain the topic
+		KafkaTestUtils.getSingleRecord(consumer, INT_KEY_TOPIC);
+		pf.destroy();
+	}
+
+	@Test
 	public void testWithCallback() throws Exception {
```

Comment:
After closer review (sorry for late feedback), that looks like with this new `send()` with the `Callback` we just replicate the logic exists with the `ListenableFuture.addCallback()`.

Can not Success and Failure Message Channels logic be based on the `ListenableFuture.addCallback()`?
Looking into `producer.send()` usage in the `KafkaTemplate` we already fully meet the requirements.

WDYT?